### ISSUE TITLE
"MyTCN" field in debug screen empty when device is restarted

### ISSUE THREAD

**Issue by ivnsch** [Co-Epi/app-ios#258]
When doing a fresh install on 2 devices with BLE enabled, the TCN generation callback of both will be executed and the TCN will be shown in the debug screen ("My TCN"), and logs. 

Restarting the app on one of the devices (from Xcode or stopping/re-opening on the device), however, will not trigger the TCN generation callback, which leaves "My TCN" field on the device empty:

- Why does this happen?
- How does this affect the contact tracing (not sure whether it's actually an issue)?
- Is there an easy fix for the debug screen to show it?

Context: 

- iOS - iOS as far as documented in the [spec](https://github.com/TCNCoalition/TCN), should generate the TCN when receiving a read request. If we restart a device and are not generating it, it means that the central isn't reading or detecting the advertisements?
- The TCN generation callback can be found in `BleAdapter`

**Comment by ivnsch** [comment]
I debugged this and found the reason for this behavior:

1) iOS - iOS uses characteristic read.
2) On iOS - iOS, TCNs are generated only when receiving a read request.
3) If the listener has a [TCN cached for the sender's device identifier](https://github.com/TCNCoalition/tcn-client-ios/blob/develop/Sources/TCNClient/Bluetooth/TCNBluetoothServiceImpl.swift#L539), it doesn't send a read request.
4) The "My TCN" displayed in the UI is set when the TCN generation closure is called.

This means that if device1 is restarted, device2 has cached its TCN and will not send a new read request. The restart causes device2 to lose the "My TCN" variable, and since the TCN generation closure is not called, the UI isn't updated.

TLDR: Only an issue of the debug screen. We could try to fix it, though not sure how, e.g. we can store the last generated TCN in preferences and restore it when the device restarts. But is this TCN always meaningful? If we're testing with more than 2 devices, there will be 2 valid last generated TCNs, so we'd have to store a list, etc.

@scottleibrand 

Edit: Maybe we can store the last one and label it as such in the debug screen. This would help debugging with 2 devices, without being confusing.

**Comment by i-schuetz** [comment]
For quicker understanding, this issue manifests the same as this one https://github.com/Co-Epi/app-ios/issues/286 (https://user-images.githubusercontent.com/67011880/85437895-4aceb900-b559-11ea-9f32-2e15cacf5453.jpg).
